pkg/operator/csidriveroperator/csioperatorclient/shared-resource.go

```diff
@@ -9,12 +9,14 @@ const (
 	SharedResourceDriverName             = "csi.sharedresource.openshift.io"
 	envSharedResourceDriverOperatorImage = "SHARED_RESOURCE_DRIVER_OPERATOR_IMAGE"
 	envSharedResourceDriverImage         = "SHARED_RESOURCE_DRIVER_IMAGE"
+	envSharedResourceDriverWebhookImage  = "SHARED_RESOURCE_DRIVER_WEBHOOK_IMAGE"
 )
 
 func GetSharedResourceCSIOperatorConfig() CSIOperatorConfig {
 	pairs := []string{
 		"${OPERATOR_IMAGE}", os.Getenv(envSharedResourceDriverOperatorImage),
 		"${DRIVER_IMAGE}", os.Getenv(envSharedResourceDriverImage),
+		"${WEBHOOK_IMAGE}", os.Getenv(envSharedResourceDriverWebhookImage),
```

Comment:
I don't see `${WEBHOOK_IMAGE}` string used anywhere in https://github.com/openshift/cluster-storage-operator/blob/master/assets/csidriveroperators/shared-resource/09_deployment.yaml

Reply:
See how `DRIVER_IMAGE` env. var value is populated in the same file, you must do the same with the webhook image.

Reply:
@jsafrane so here we are bit different than the DRIVER_IMAGE. let me explain
The cluster storage operator is installing the shared resource driver operator, and hence we declare the deployment of the sro into the assets of the cso 
The webhook in an other hand, is installed via the shared resource operator, and its deployment is not known by the cluster storage operator. so Yes I am adding the ${WEBHOOK_IMAGE} string in order to propagate the image of the webhook referenced in the cluster storage operator to the shared resource operator https://github.com/openshift/csi-driver-shared-resource-operator/pull/45/files#diff-f7f6af5f4443f2edda1e6b8ed34f9ac83e4e087eb4df60b345c1437bd2e9466eR20

I am,  may be mistaken the way to propagate the env var, so please correct me if I am wrong

Reply:
@jkhelil I'm not sure I see what you are getting at, but even though the webhook will be a deployment instead of a daemonset like the driver, the separation of concern wrt this between the cluster storage operator, shared resource operator, and the shared resource repo should be following the same pattern

your PR 45 should have in its deployment something similar to https://github.com/openshift/csi-driver-shared-resource-operator/blob/master/assets/node.yaml#L53 for the image, where it leverages the env var we are propagating from CSO to SRO etc.

Reply:
and that is what I see at the https://github.com/openshift/csi-driver-shared-resource-operator/pull/45/files#diff-f7f6af5f4443f2edda1e6b8ed34f9ac83e4e087eb4df60b345c1437bd2e9466eR20 item you referenced

I believe @jsafrane corrections are what will make that flow work

Reply:
@gabemontero Thank you for the clarification, I think I am getting distured with the cascade of opeartors, but yes you right no difference between DRIVER and WEBHOOK
and here is what is missing @jsafrane : Adding the WEBHOOK_IMAGE string as env var of the shared resource driver operator